File: fixtures/scale/ruby/large/lib/check_shiftreel.rb
require_relative 'shiftreel_encrypt'
require_relative 'shiftreel_decrypt'
require_relative 'corpus_additive'

def check_shiftreel
  corpus_additive.each do |text|
    sealed = shiftreel_encrypt(text, 6)
    opened = shiftreel_decrypt(sealed, 6)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_shiftreel.py
from cipherlab.ciphers.shiftreel import shiftreel_encrypt, shiftreel_decrypt
from cipherlab.specs.shiftreel_spec import shiftreel_spec
from cipherlab.corpus.corpus_additive import corpus_additive


def check_shiftreel():
    spec = shiftreel_spec()
    for text in corpus_additive():
        sealed = shiftreel_encrypt(text, spec["key"])
        opened = shiftreel_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "additive"
